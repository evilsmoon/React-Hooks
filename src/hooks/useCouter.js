import { useState } from 'react';

export const useCouter = (initialState=10) => {

    const [state, setstate] = useState(initialState);
    const increment = (factor= 1)=>{
        setstate(state+factor);
    }
    const reset = ()=>{
        setstate(initialState)
    }
    const decrement = (factor= 1)=>{
        setstate(state-factor);
    }

    return {
        state,
        increment,
        reset,
        decrement
    };

}
