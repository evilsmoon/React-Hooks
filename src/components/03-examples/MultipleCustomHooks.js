import React from 'react';
import { useCouter } from '../../hooks/useCouter';
import { useFetch } from '../../hooks/useFetch';
import './examples.css';
export const MultipleCustomHooks = () => {

    const {counter,increment}= useCouter(1);
    const { loading ,data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`)

    const {author,quote} = !!data && data[0]
    console.log(author,quote)
    
    
    return (
        <div>
            <h1>Breaking Bad || Custom Hooks!!! </h1>
            <hr />

            {
                loading
                    ?
                    (
                        <div className='alert alert-info text-center'>
                            loading...
                        </div>
                    )
                    :
                    (
                        <blockquote className='blockquote text-right'>
                            <p className='mb-0'>{quote}</p>
                            <footer className='blockquote-footer'>{author}</footer>
                        </blockquote>
                    )
            }
        <button className='btn btn-info' onClick={increment}>
                Next Quote

        </button>

        </div>
    )
}
