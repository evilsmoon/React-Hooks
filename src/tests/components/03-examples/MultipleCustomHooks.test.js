import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCouter } from '../../../hooks/useCouter';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCouter');

describe('test MultipleCustomHooks.js', () => {


    useCouter.mockReturnValue({
        counter:10,
        increment: ()=>{}
    })
    test('debe de mostrat correctamente', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar informacion', () => {

        useFetch.mockReturnValue({

            data: [{ 
                author: 'gabriel', 
                quote: 'hii ;v' 
            }],
            loading: false,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        // expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-0').text().trim() ).toBe( 'hii ;v' );
        expect( wrapper.find('footer').text().trim() ).toBe( 'gabriel' );
    });

})
