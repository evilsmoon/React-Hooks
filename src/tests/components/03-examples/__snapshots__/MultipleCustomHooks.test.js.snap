// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MultipleCustomHooks.js debe de mostrat correctamente 1`] = `
<div>
  <h1>
    Breaking Bad || Custom Hooks!!! 
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-info"
    onClick={[Function]}
  >
    Next Quote
  </button>
</div>
`;
